plutus-ledger-api/src/Plutus/ApiCommon.hs

```diff
@@ -1,23 +1,54 @@
+{-# LANGUAGE DeriveAnyClass     #-}
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE TypeApplications   #-}
 {-|
 Common types and functions used across all the ledger API modules.
 -}
 module Plutus.ApiCommon where
 
+import Codec.CBOR.Decoding qualified as CBOR
+import Codec.CBOR.Extras
+import Codec.CBOR.Read qualified as CBOR
+import Control.DeepSeq
+import Control.Monad.Except
+import Control.Monad.Writer
+import Data.Bifunctor
+import Data.ByteString.Lazy (fromStrict)
 import Data.ByteString.Short
+import Data.Either
 import Data.Foldable (fold)
 import Data.Map qualified as Map
 import Data.Set qualified as Set
 import Data.Text
+import Data.Tuple
+import GHC.Base
+import GHC.Generics
 import PlutusCore as PLC
+import PlutusCore.Data as PLC
+import PlutusCore.Evaluation.Machine.ExBudget (ExBudget (..))
+import PlutusCore.Evaluation.Machine.ExBudget qualified as PLC
+import PlutusCore.Evaluation.Machine.MachineParameters as Plutus
+import PlutusCore.MkPlc qualified as PLC
+import PlutusCore.Pretty
+import PlutusPrelude (through)
 import Prettyprinter
+import UntypedPlutusCore qualified as UPLC
+import UntypedPlutusCore.Check.Scope qualified as UPLC
+import UntypedPlutusCore.Evaluation.Machine.Cek (CekValue)
+import UntypedPlutusCore.Evaluation.Machine.Cek qualified as UPLC
 
 -- | This represents the Cardano protocol version, with its major and minor components.
 -- This relies on careful understanding between us and the ledger as to what this means.
 data ProtocolVersion = ProtocolVersion { pvMajor :: Int, pvMinor :: Int }
   deriving stock (Show, Eq)
 
+type LanguageVersion = PLC.Version ()
```

Comment:
Again, I think we shouldn't use this type. It's a specific enum corresponding to specific things in the ledger, it's not related to `Version` at all really.

Maybe
```
data LedgerPlutusVersion = PlutusV1 | Plutus V2
```

Reply:
that's what i wanted to talk to you about on the chat about deprecating plc.version!

Reply:
Anyway ledger has such  an enum as well, we should perhaps expose it in apicommon and then we can both use the same.

Reply:
https://github.com/input-output-hk/cardano-ledger/blob/dec42f4075568347da53db318c0658bcc5267e5f/eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Language.hs#L24-L27